pkg/tfbridge/walk/walk_test.go

```diff
@@ -0,0 +1,201 @@
+		Schema: schema.SchemaMap{
+			"x_prop": strSchema,
+			"y_prop": strSchema,
+		},
+	}).Shim()
+
+	schemaMap := &schema.SchemaMap{
+		"string_prop": strSchema,
+		"list_str": (&schema.Schema{
+			Type: shim.TypeList,
+			Elem: strSchema,
+		}).Shim(),
+		"list_unknowns": (&schema.Schema{
+			Type: shim.TypeList,
+		}).Shim(),
+		"flat_list": (&schema.Schema{
+			Type:     shim.TypeList,
+			Elem:     xySchema,
+			MaxItems: 1,
+		}).Shim(),
+		"obj_set": (&schema.Schema{
+			Type: shim.TypeSet,
+			Elem: xySchema,
+		}).Shim(),
+		"single_obj": (&schema.Schema{
+			Type: shim.TypeMap,
+			Elem: xySchema,
+		}).Shim(),
+	}
+
+	type testCase struct {
+		name        string
+		pp          resource.PropertyPath
+		schemaInfos map[string]*tfbridge.SchemaInfo
+		expected    walk.SchemaPath
+	}
+
+	cases := []testCase{
+		{
+			name:     "simple string property",
+			pp:       []any{"stringProp"},
+			expected: walk.NewSchemaPath().GetAttr("string_prop"),
+		},
+		{
+			name:     "simple not found property",
+			pp:       []any{"notFoundProp"},
+			expected: nil,
+		},
+		{
+			name:     "simple not found drill-down property",
+			pp:       []any{"stringProp", "notFoundProp"},
+			expected: nil,
+		},
+		{
+			name:     "list",
+			pp:       []any{"listStr"},
+			expected: walk.NewSchemaPath().GetAttr("list_str"),
+		},
+		{
+			name:     "list element",
+			pp:       []any{"listStr", 3},
+			expected: walk.NewSchemaPath().GetAttr("list_str").Element(),
+		},
+		{
+			name:     "list of unknowns",
+			pp:       []any{"listUnknowns"},
+			expected: walk.NewSchemaPath().GetAttr("list_unknowns"),
+		},
+		{
+			name:     "list of unknowns element",
+			pp:       []any{"listUnknowns", 3},
+			expected: nil,
+		},
+		{
+			name:     "single-nested block",
+			pp:       []any{"singleObj", "xProp"},
+			expected: walk.NewSchemaPath().GetAttr("single_obj").GetAttr("x_prop"),
+		},
+		{
+			name:     "set-nested block 1",
+			pp:       []any{"objSet"},
+			expected: walk.NewSchemaPath().GetAttr("obj_set"),
+		},
+		{
+			name:     "set-nested block 2",
+			pp:       []any{"objSet", 0},
+			expected: walk.NewSchemaPath().GetAttr("obj_set").Element(),
+		},
+		{
+			name:     "set-nested block 3",
+			pp:       []any{"objSet", 0, "xProp"},
+			expected: walk.NewSchemaPath().GetAttr("obj_set").Element().GetAttr("x_prop"),
+		},
+		{
+			name:     "max-items-1 list 1",
+			pp:       []any{"flatList"},
+			expected: walk.NewSchemaPath().GetAttr("flat_list"),
+		},
+		{
+			name:     "max-items-1 list 3",
+			pp:       []any{"flatList", "xProp"},
+			expected: walk.NewSchemaPath().GetAttr("flat_list").Element().GetAttr("x_prop"),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			pp := PropertyPathToSchemaPath(tc.pp, schemaMap, tc.schemaInfos)
+			assert.Equal(t, tc.expected, pp)
+		})
+	}
+}
+
+func TestLookupSchemaInfoMapPath(t *testing.T) {
+	yes := true
+
+	schemaInfos := map[string]*tfbridge.SchemaInfo{
```

Comment:
Can we have a test case for when `SchemaInfos` declares something `MaxItemsOne: &true`.